lib/client/redirect.go

```diff
@@ -202,6 +223,11 @@ func (rd *Redirector) callback(w http.ResponseWriter, r *http.Request) (*auth.SS
 		return nil, trace.NotFound("path not found")
 	}
 
+	if r.URL.Query().Has("err") {
+		err := r.URL.Query().Get("err")
+		return nil, trace.Errorf("received error: %v", err)
```

Comment:
nit: `trace.Errorf` isn't generally used directly
```suggestion
		return nil, trace.Wrap(err)
```

Reply:
No can do: `err` is string here.

Reply:
Is this a user facing error? If so, can the error message be made a bit clearer? E.g. "identity provider callback failed with the following error" or "error while handling the identity provider callback", or something.